### PR TITLE
Fixes GO inline array generation for uuid

### PR DESCRIPTION
**Pull request by rkodev**
Resolves https://github.com/microsoftgraph/microsoft-graph-docs/issues/21282
Resolves https://github.com/microsoftgraph/microsoft-graph-docs/issues/21304

Fixes 

- [x] Inline UUID script for arrays
- [x] TypeInformation for slices / arrays when generation SnippetCodeGraph
- [x] Adding the `Item` naming scheme in go Me/Users Package 

Current example https://learn.microsoft.com/en-us/graph/api/user-assignlicense?view=graph-rest-1.0&tabs=go

```go

import (
	  "context"
	  msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
	  graphmodels "github.com/microsoftgraph/msgraph-sdk-go/Me/AssignLicense"
	  //other-imports
)

graphClient := msgraphsdk.NewGraphServiceClientWithCredentials(cred, scopes)


requestBody := graphmodels.NewAssignLicensePostRequestBody()


assignedLicense := graphmodels.NewAssignedLicense()
disabledPlans := []string {
 := uuid.MustParse("11b0131d-43c8-4bbb-b2c8-e80f9a50834a")
assignedLicense.Set(&) 

}
assignedLicense.SetDisabledPlans(disabledPlans)
skuId := uuid.MustParse("45715bb8-13f9-4bf6-927f-ef96c102d394")
assignedLicense.SetSkuId(&skuId) 

addLicenses := []graphmodels.AssignedLicenseable {
	assignedLicense,

}
requestBody.SetAddLicenses(addLicenses)
removeLicenses := []string {
 := uuid.MustParse("bea13e0c-3828-4daa-a392-28af7ff61a0f")
requestBody.Set(&) 

}
requestBody.SetRemoveLicenses(removeLicenses)

result, err := graphClient.Me().AssignLicense().Post(context.Background(), requestBody, nil)
```

Fixed result

```go
import (
	  "context"
	  msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
	  graphusers "github.com/microsoftgraph/msgraph-sdk-go/users"
	  graphmodels "github.com/microsoftgraph/msgraph-sdk-go/models"
	  //other-imports
)

graphClient, err := msgraphsdk.NewGraphServiceClientWithCredentials(cred, scopes)


requestBody := graphusers.NewItemAssignLicensePostRequestBody()


assignedLicense := graphmodels.NewAssignedLicense()
disabledPlans := []uuid.UUID {
	uuid.MustParse("11b0131d-43c8-4bbb-b2c8-e80f9a50834a"),
}
assignedLicense.SetDisabledPlans(disabledPlans)
skuId := uuid.MustParse("45715bb8-13f9-4bf6-927f-ef96c102d394")
assignedLicense.SetSkuId(&skuId) 

addLicenses := []graphmodels.AssignedLicenseable {
	assignedLicense,
}
requestBody.SetAddLicenses(addLicenses)
removeLicenses := []uuid.UUID {
	uuid.MustParse("bea13e0c-3828-4daa-a392-28af7ff61a0f"),
}
requestBody.SetRemoveLicenses(removeLicenses)

result, err := graphClient.Me().AssignLicense().Post(context.Background(), requestBody, nil)

```

### CODE REVIEW

File: CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs [hunk 1]
```diff
@@ -461,13 +461,29 @@ private static CodeProperty parseJsonArrayValue(string propertyName, JsonElement
         {
             var alternativeType = schema?.Items?.AnyOf?.FirstOrDefault()?.AllOf?.LastOrDefault()?.Title;
             var children = value.EnumerateArray().Select(item => parseProperty(schema.GetSchemaTitle() ?? alternativeType?.ToFirstCharacterUpperCase(), item, schema?.Items)).ToList();
+
+            // uuid schemas 
             var genericType = schema.GetSchemaTitle().ToFirstCharacterUpperCase() ??
                               (value.EnumerateArray().Any() ?
-                                  value.EnumerateArray().First().ValueKind.ToString() :
+                                  evaluatePropertyTypeDefinition(value.EnumerateArray().First().ValueKind.ToString(), schema?.Items) :
```
Comment (andrueastman) [inline CodeSnippetsReflection.OpenAPI/ModelGraph/SnippetCodeGraph.cs:468]: Should we also consider propagating the typedefinition infomation to the nested children if any?

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@
 using CodeSnippetsReflection.StringExtensions;
 using Microsoft.OpenApi.Services;
 using System.Text.RegularExpressions;
+using System.Xml;
```
Comment (andrueastman) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:10]: Unused import

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs [hunk 3]
```diff
@@ -345,7 +355,9 @@ private static void WriteBody(SnippetCodeGraph codeGraph, StringBuilder builder)
             }
             else
             {
-                builder.AppendLine($"{indentManager.GetIndent()}{requestBodyVarName} := graph{ProcessFinalNameSpaceName(codeGraph.Body.NamespaceName).Replace(".","").ToLowerInvariant()}.New{codeGraph.Body.Name.ToFirstCharacterUpperCase()}()");
+                // objects in namespace user have a prefix of item
+                string bodyName = codeGraph.Body.NamespaceName.StartsWith("Me") ? $"Item{codeGraph.Body.Name}" : codeGraph.Body.Name;
```
Comment (baywet) [inline CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:359]: String comparison for starts with

File: CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs (hunk 2 above)
Comment (rkodev) [reply CodeSnippetsReflection.OpenAPI/LanguageGenerators/GoGenerator.cs:10]: Resolved

**Review by andrueastman — APPROVED** [review]

**Review by baywet — APPROVED** [review]

**Pull request merged**